spark/src/main/scala/org/apache/comet/rules/RewriteJoin.scala

```diff
@@ -67,4 +83,21 @@ object RewriteJoin extends JoinSelectionHelper {
       }
     case _ => plan
   }
+
+  def getOptimalBuildSide(join: Join): BuildSide = {
+    val leftSize = join.left.stats.sizeInBytes
+    val rightSize = join.right.stats.sizeInBytes
+    val leftRowCount = join.left.stats.rowCount
+    val rightRowCount = join.right.stats.rowCount
+    if (leftSize == rightSize && rightRowCount.isDefined && leftRowCount.isDefined) {
```

Comment:
why checking the sizes matters? 🤔 shouldn't be it enough to use rowcounts?

Reply:
rowCount is an `Option` so may not always be available. Row count is only used here as a tie-breaker if the left and right size are the same

Reply:
We can also revisit this logic as part of https://github.com/apache/datafusion-comet/issues/1430

Reply:
maybe I missing something? `leftSize == rightSize` condition looks very unlikely so by the logic it would never consider rowCounts here and fallback to sizes. 

We can perhaps use something like https://docs.pingcap.com/tidb/stable/join-reorder#example-the-greedy-algorithm-of-join-reorder

and if rowCounts are not available fallback to sizes

Reply:
I suppose it depends on the goal for choosing the build-side. Do we want to limit the amount of data that needs to be loaded into the hash map or limit the number of rows?

The article you linked is for reordering nested joins, so row count would likely be more critical for that scenario.

Reply:
oh this is for build side, not the entire join reorder, that was I missing. For hash join build side it is a rule of thumb of having smaller table for build side to fit in memory, but what means smaller here I dont have a strong opinion. Perhaps if we talk about memory then sizeInBytes makes more sense.